rewrite-java-test/src/test/java/org/openrewrite/java/service/AnnotationServiceTest.java

```diff
@@ -190,4 +190,117 @@ public J.FieldAccess visitFieldAccess(J.FieldAccess fieldAccess, Integer integer
+          )
+        );
+    }
+
+    @Test
+    void classAnnotationFromInterface() {
+        rewriteRun(
+          java(
+            """
+              import java.lang.annotation.*;
+
+              @Retention(RetentionPolicy.RUNTIME)
+              @Target(ElementType.TYPE)
+              @interface InterfaceAnnotation {}
+
+              @InterfaceAnnotation
+              interface MyInterface {}
+
+              class ImplementingClass implements MyInterface {}
+              """,
+            spec -> spec.afterRecipe(cu -> new JavaIsoVisitor<Integer>() {
+                @Override
+                public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, Integer p) {
+                    AnnotationService service = service(AnnotationService.class);
+                    if ("ImplementingClass".equals(classDecl.getSimpleName())) {
+                        assertThat(service.isAnnotatedWith(classDecl, "InterfaceAnnotation")).isTrue();
+                        assertThat(service.annotatedWith(classDecl, "InterfaceAnnotation")).hasSize(1);
+                    }
+                    return classDecl;
+                }
+            }.visit(cu, 0))
+          )
+        );
+    }
+
+    @Test
+    void methodAnnotationInheritance() {
+        rewriteRun(
+          java(
+            """
+              import java.lang.annotation.*;
+
+              @Retention(RetentionPolicy.RUNTIME)
+              @Target(ElementType.METHOD)
+              @interface MethodAnnotation {}
+
+              class BaseClass {
+                  @MethodAnnotation
+                  public void doSomething() {}
+              }
+
+              class ChildClass extends BaseClass {
+                  @Override
+                  public void doSomething() {}
+              }
+              """,
+            spec -> spec.afterRecipe(cu -> new JavaIsoVisitor<Integer>() {
+                @Override
+                public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Integer p) {
+                    AnnotationService service = service(AnnotationService.class);
+                    Cursor parentCursor = getCursor().getParentTreeCursor();
+                    if (parentCursor.getValue() instanceof J.ClassDeclaration) {
+                        J.ClassDeclaration classDecl = parentCursor.getValue();
+                        if ("ChildClass".equals(classDecl.getSimpleName()) &&
+                            "doSomething".equals(method.getSimpleName())) {
+                            // Direct getAllAnnotations should only show @Override
+                            List<J.Annotation> directAnnotations = service.getAllAnnotations(getCursor());
+                            assertThat(directAnnotations).hasSize(1);
+                            assertThat(directAnnotations.get(0).getSimpleName()).isEqualTo("Override");
+
+                            // But hasAnnotation should find @MethodAnnotation from parent
```

Comment:
Looks great, note the comment says `hasAnnotation` but the actual method is `isAnnotatedWith`. This is my fault I suggested both names in the issue. 8) 
The previous test has a similar comment

Reply:
It's my fault for being indecisive about what to call it. Did it one way, then changed it to the other